frontend/src/Notes.tsx

```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import {NotesRepository} from "./NotesRepository";
+
+type NotesProps = {
+  notesRepository: NotesRepository
```

Comment:
React doesn't have an opinion about dependency injection. Here you use the traditional way. This makes the component context independent since it gets everything it needs through the props.
Usually components are split into two categories: "Presentational" and "Container" components (names may vary).

Presentational components are simple, they define the data they need in props, take it and just render it in some manner. E.g. a button with text string and onClick callback as props.

Container components have logic and fetching of data and uses either zero, one or many presentational components to render stuff. This makes container components more tied to a specific use case keeping presentational ones easily reusable. Services (or repositories) are usually not injected in this traditional way because they can be easily mocked without it.